File: medAndro/10845.py
import sys
input = sys.stdin.readline
print = sys.stdout.write

n = int(input().rstrip())
queue = []
len = 0
for i in range(n):
    cmd= input().rstrip()
    if "push" in cmd:
        queue.append(cmd.split()[1])
        len +=1
    elif "pop" == cmd:
        if len == 0:
            print("%s\n" % -1)
        else:
            print("%s\n" % queue.pop(0))
            len -= 1
    elif "size" == cmd:
        print("%s\n" % len)
    elif "empty" == cmd:
        if len == 0:
            print("%s\n" % 1)
        else:
            print("%s\n" % 0)
    elif "front" == cmd:
        if len == 0:
            print("%s\n" % -1)
        else:
            print("%s\n" % queue[0])
    elif "back" == cmd:
        if len == 0:
            print("%s\n" % -1)
        else:
            print("%s\n" % queue[len-1])
    else:
        if len == 0:
            print("%s\n" % -1)
        else:
            print("%s\n" % queue[len-1])

File: dhkim/10845.rs
use std::collections::LinkedList;
use std::io;
use std::fmt::Write;
use std::str::SplitWhitespace;

fn op(cmd: &mut SplitWhitespace, queue: &mut LinkedList<usize>) -> String {
    let mut output = String::new();
    match cmd.next().unwrap() {
        "push" => {
            queue.push_back(cmd.next().unwrap().parse::<usize>().unwrap())
        },
        "pop" => {
            match queue.pop_front() {
                Some(i) => {
                    writeln!(output, "{i}").unwrap();
                },
                None => {
                    writeln!(output, "{}", -1).unwrap();
                }
            }
        },
        "size" => {
            writeln!(output, "{}", queue.len()).unwrap();
        },
        "empty" => {
            writeln!(output, "{}", if queue.is_empty() { 1 } else { 0 }).unwrap();
        },
        "front" => {
            match queue.front() {
                Some(i) => {
                    writeln!(output, "{i}").unwrap();
                },
                None => {
                    writeln!(output, "{}", -1).unwrap();
                }
            }
        },
        "back" => {
            match queue.back() {
                Some(i) => {
                    writeln!(output, "{i}").unwrap();
                },
                None => {
                    writeln!(output, "{}", -1).unwrap();
                }
            }
        },
        _ => { }
    }

    output
}

fn main() {
    let mut input = String::new();
    let mut output = String::new();

    io::stdin().read_line(&mut input).unwrap();
    let n: usize = input.trim().parse().unwrap();

    let mut queue = LinkedList::new();
    for _ in 0..n {
        input.clear();
        io::stdin().read_line(&mut input).unwrap();
        let mut cmd = input.split_whitespace();
        write!(output, "{}", op(&mut cmd, &mut queue)).unwrap();
    }

    println!("{output}");
}
